src/models/hit-types/tv-clip-hit.ts

```diff
@@ -0,0 +1,237 @@
+ *
+ * The fields in here are cast to their respective field types. See `metadata-fields/field-type`.
+ *
+ * Add additional fields as needed.
+ *
+ * @export
+ * @class TvClipHit
+ */
+export class TvClipHit {
+  /**
+   * This is the raw hit response; useful for inspecting the raw data
+   * returned from the server.
+   */
+  rawMetadata?: Record<string, any>;
+
+  constructor(json: Record<string, any>) {
+    this.rawMetadata = json;
+  }
+
+  /**
+   * The item identifier.
+   *
+   * _Note_: This is a plain string instead of a `MetadataField` since it's
+   * the primary key of the item.
+   */
+  get identifier(): typeof Metadata.prototype.identifier {
+    return this.rawMetadata?.fields?.identifier;
+  }
+
+  /**
+   * Synthesized in processing of TVS API hit; TBD
+   * Optional.
+   */
+  @Memoize() get highlight(): StringField | undefined {
+    // Note: _not_ inside the fields object.
+    return this.rawMetadata?.highlight?.text
+      ? new StringField(this.rawMetadata.highlight.text)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get addeddate(): typeof Metadata.prototype.addeddate {
+    return this.rawMetadata?.fields?.addeddate
+      ? new DateField(this.rawMetadata.fields.addeddate)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get avg_rating(): typeof Metadata.prototype.avg_rating {
+    return this.rawMetadata?.fields?.avg_rating != null
+      ? new NumberField(this.rawMetadata.fields.avg_rating)
+      : undefined;
+  }
+
+  /** Multivalued. */
+  @Memoize() get collection(): typeof Metadata.prototype.collection {
+    return this.rawMetadata?.fields?.collection
+      ? new StringField(this.rawMetadata.fields.collection)
+      : undefined;
+  }
+
+  @Memoize() get created_on(): DateField | undefined {
+    return this.rawMetadata?.fields?.created_on
+      ? new DateField(this.rawMetadata.fields.created_on)
+      : undefined;
+  }
+
+  /**
+   * Optional.
+   * Multivalued.
+   */
+  @Memoize() get creator(): typeof Metadata.prototype.creator {
+    return this.rawMetadata?.fields?.creator
+      ? new StringField(this.rawMetadata.fields.creator)
+      : undefined;
+  }
+
+  /** Optional. */
+  @Memoize() get date(): typeof Metadata.prototype.date {
+    return this.rawMetadata?.fields?.date
+      ? new DateField(this.rawMetadata.fields.date)
+      : undefined;
+  }
+
+  /**
+   * Contents of description in TVS API hit, TBD
+   * Optional.
+   * Multivalued.
+   */
+  @Memoize() get description(): typeof Metadata.prototype.description {
+    return this.rawMetadata?.fields?.description
+      ? new StringField(this.rawMetadata.fields.description)
+      : undefined;
+  }
+
+  /**
+   * Total views over ITEM (not text) lifetime, updated by audit consultation with Views API.
+   * Optional.
+   */
+  @Memoize() get downloads(): typeof Metadata.prototype.downloads {
+    return this.rawMetadata?.fields?.downloads != null
```

Comment:
SUGGESTION: Here and below, shouldn't we always prefer explicit comparisons without type coercion, i.e. `!==` over `!=`?

Reply:
Happy to do that! The `!=` is also used for all the other hit types, so I'm happy to switch that out there as well.